include/mbedtls/debug.h

```diff
@@ -112,6 +112,11 @@ void mbedtls_debug_set_threshold( int threshold );
  * \attention       This function is intended for INTERNAL usage within the
  *                  library only.
  */
+#if defined(__GNUC__) || defined(__clang__)
+#if __has_attribute(format)
```

Comment:
I think we need
```
#if defined(__has_attribute)
#if __has_attribute(format)
```
here. Otherwise `armcc --gnu` fails with:
```
% armcc --gnu -c attr.c                                                                                                                                    git|development|
"attr.c", line 2: Error:  #59: function call is not allowed in a constant expression
  #if __has_attribute(format)
      ^
attr.c: 0 warnings, 1 error
```
(and similarly if we try `defined(__has_attribute) && __has_attribute(format)`)

Reply:
Thanks @mpg for pointing this out. However, what's wrong with `defined(__has_attribute) && __has_attribute(format)`?

Reply:
Ok, I thought the preprocessor wouldn't evaluate the second argument if `defined(__has_attribute)` is false - but that doesn't seem to be the case.

Reply:
I think the problem is that even if the preprocessor doesn't _evaluate_ the second argument, it needs to be able to _parse_ it, and I guess it can only do that if it knows whether a given symbol is a function-like macro or not (and its arity if it is). While with mutliple lines, the preprocessor will not even attempt to parse ignored lines (beyond checking `#if` nesting). At least that's how I think it works.

Oh well, I went ahead and actually checked the standard, I wasn't too far:
```
4 When in a group that is skipped (6.10.1), the directive syntax is relaxed to allow any
sequence of preprocessing tokens to occur between the directive name and the following
new-line character.
```

Reply:
@mpg Yes, that makes sense, thank you!